### PR TITLE
feat: refactor dymint config to support loading configuration from file

### PR DESCRIPTION
**Pull request by mtsitrin**
* Main change is the breaking of the SL config from raw json to an explicit struct.
* also changing the `namespaceID` to string so it will be easier to set in conf file
* renamed `settlement.LayerClient` to `settlement.LayerI` to indicate it's an interface

# PR Standards

## Opening a pull request should be able to meet the following requirements

---

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

File: settlement/config.go [hunk 1]
```diff
@@ -1,17 +1,14 @@
 package settlement
 
-import "github.com/ignite/cli/ignite/pkg/cosmosaccount"
-
 // Config for the DymensionLayerClient
 type Config struct {
-	KeyringBackend cosmosaccount.KeyringBackend `json:"keyring_backend"`
```
Comment (omritoptix) [inline settlement/config.go]: why did you remove `KeyringBackend`?

File: conv/config_test.go [hunk 2]
```diff
@@ -12,14 +12,18 @@ import (
 func TestGetNodeConfig(t *testing.T) {
 	t.Parallel()
 
+	validCosmos := "127.0.0.1:1234"
+	validOptimint := "/ip4/127.0.0.1/tcp/1234"
```
[the remark this answers is not in the document]
Comment (mtsitrin) [reply conv/config_test.go:16]: copied it from `conv/addr_test.go`

File: settlement/config.go (hunk 1 above)
Comment (mtsitrin) [reply settlement/config.go]: we only support `test` for the sequencer
no reason to expose it as a configuration parameter if it's not supported

File: state/executor.go [hunk 3]
```diff
@@ -37,17 +38,23 @@ type BlockExecutor struct {
 
 // NewBlockExecutor creates new instance of BlockExecutor.
 // Proposer address and namespace ID will be used in all newly created blocks.
-func NewBlockExecutor(proposerAddress []byte, namespaceID [8]byte, chainID string, mempool mempool.Mempool, proxyApp proxy.AppConns, eventBus *tmtypes.EventBus, logger log.Logger) *BlockExecutor {
-	return &BlockExecutor{
+func NewBlockExecutor(proposerAddress []byte, namespaceID string, chainID string, mempool mempool.Mempool, proxyApp proxy.AppConns, eventBus *tmtypes.EventBus, logger log.Logger) *BlockExecutor {
+	bytes, err := hex.DecodeString(namespaceID)
+	if err != nil {
+		panic(err)
```
[the remark this answers is not in the document]
Comment (mtsitrin) [reply state/executor.go:44]: i'll add some logging

File: settlement/config.go (hunk 1 above)
Comment (omritoptix) [reply settlement/config.go]: are you sure? In the cosmosclient we init I see this keyring backends supported:
```
type KeyringBackend string

const (
	// KeyringTest is the test keyring backend. With this backend, your keys will be
	// stored under your app's data dir,
	KeyringTest KeyringBackend = "test"

	// KeyringOS is the OS keyring backend. with this backend, your keys will be
	// stored in your operating system's secured keyring.
	KeyringOS KeyringBackend = "os"

	// KeyringMemory is in memory keyring backend, your keys will be stored in application memory.
	KeyringMemory KeyringBackend = "memory"
)
```

either way we'll for sure need to support more keyring backends in the near future (specifically ledger) so not sure what's the motivation for removing it

Comment (mtsitrin) [reply settlement/config.go]: yeah but the sequencer can't use anything else as it will request signing approval for each submission.
anyway I leave it in the configuration

**Review by omritoptix — DISMISSED** [review]

**Review by omritoptix — APPROVED** [review]
